shaders/src/depth_main.vs

```diff
@@ -3,15 +3,17 @@
 void materialVertex(inout MaterialVertexInputs m) { }
 
 void main() {
-#if defined(VERTEX_DOMAIN_DEVICE)
-    gl_Position = getPosition();
-#else
     MaterialVertexInputs material;
     initMaterialVertex(material);
     materialVertex(material);
+#if defined(VERTEX_DOMAIN_DEVICE)
+    gl_Position = getPosition();
+#else
     gl_Position = getClipFromWorldMatrix() * getWorldPosition(material);
 #endif
 
+    vertex_worldPosition = material.worldPosition.xyz;
```

Comment:
This will only be used by VSM variants, should it be #if'd?

Reply:
That would add another depth variant, which might be acceptable to avoid computing this (and the `getPosition()` above).

Reply:
Actually, this would add 2 new depth variants. I'll go ahead and add the variants to make this conditional.. we can always remove the variants in the future without breaking materials if we decide to pay the computational cost instead.